### ISSUE TITLE
Exception thrown when a function is used for the error content of the observe method

### ISSUE THREAD

**Issue by orestisioakeimidis** [ngneat/hot-toast#153]
## I'm submitting a...

<pre><code>
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

When a function is used for the error content of the `HotToastService.observe` method, the following error is thrown in the console.
```
ASSERTION ERROR: Provided Component class doesn't contain Component definition. Please check whether provided class has @Component decorator. [Expected=> null != null <=Actual]
```
<img width="1282" alt="Screenshot 2024-01-29 at 22 58 45" src="https://github.com/ngneat/hot-toast/assets/5999499/df0f9d3f-bbad-42f2-bf2d-db1ef68fb4d9">

## Expected behavior

The error content should be shown without any exceptions thrown.

## Environment

<pre><code>
Angular version: 17.1.0
@ngneat/hot-toast: 6.1.0
@ngneat/overview: 5.1.1

Browser:
- [x] Chrome (desktop) version 120

</code></pre>

**Comment by Rohit1024** [comment]
Yes Same error observed for me while using `toast.observe` here's my minimal implementation : 
### Error : 
```
ERROR Error: ASSERTION ERROR: Provided Component class doesn't contain Component definition. Please check whether provided class has @Component decorator. [Expected=> null != null <=Actual]
    at throwError (core.mjs:393:11)
    at assertDefined (core.mjs:389:9)
    at ViewContainerRef.createComponent (core.mjs:19944:17)
    at new CompRef (ngneat-overview.mjs:108:36)
    at ViewService.createComponent (ngneat-overview.mjs:268:16)
    at DynamicViewDirective.resolveContentType (ngneat-overview.mjs:344:45)
    at DynamicViewDirective.ngOnChanges (ngneat-overview.mjs:325:18)
    at DynamicViewDirective.rememberChangeHistoryAndInvokeOnChangesHook (core.mjs:2510:14)
    at callHookInternal (core.mjs:3544:14)
    at callHook (core.mjs:3575:9)
```
#### login.component.ts : 
```ts
import { CommonModule } from '@angular/common';
import { Component, inject } from '@angular/core';
import { HotToastService } from '@ngneat/hot-toast';
import { AuthService } from '@services/auth.service';

@Component({
  standalone: true,
  selector: 'app-auth',
  imports: [CommonModule],
  template: `
    <button (click)="login()">Login</button>`,
})
export class AuthComponent {
  authService = inject(AuthService);
  private toast = inject(HotToastService);
  user$ = this.authService.currentUser$;

  login() {
    this.authService
      .login({
        email: 'xyz@email.com',
        password: 'qwerty',
      })
      .pipe(
        this.toast.observe({
          loading: 'Signing In...',
          success: (credential) =>
            `Sucessfully signed in : ${credential.user.email}`,
          error: (error) => `Error Occurred ${error.code}`,
        })
      )
      .subscribe();
  }
}
```
Where I have configured the hot-toast as follows : 
#### app.config.ts : 
```ts
  import { AppComponent } from './src/app.component';

  import { provideHotToastConfig } from '@ngneat/hot-toast';

  bootstrapApplication(AppComponent, {
    providers: [
      provideHotToastConfig(),    
     ]
  });
```
I've tried to debug this error through Angular dev tools but no luck 👀

**Comment by baikov-ilia** [comment]
Yep, seems like @ngneat/overview dependency doesn't work as expected with Angular 17

**Comment by shhdharmen** [comment]
Can you create a stackblitz to showcase the issue? And can you see if updating `@ngneat/overview` to `6.0.0`?

**Comment by baikov-ilia** [comment]
> Can you create a stackblitz to showcase the issue? And can you see if updating `@ngneat/overview` to `6.0.0` resolves the issue?

```
No matching export in "node_modules/@ngneat/overview/fesm2022/ngneat-overview.mjs" for import "DynamicViewModule"

    node_modules/@ngneat/hot-toast/fesm2022/ngneat-hot-toast.mjs:6:37:
      6 │ import { isTemplateRef, isComponent, DynamicViewModule } from '@ngneat/overview';
        ╵                                      ~~~~~~~~~~~~~~~~~

D:\Angular\cloud\node_modules\vite\node_modules\esbuild\lib\main.js:1651
  let error = new Error(text);
              ^

Error: Build failed with 1 error:
node_modules/@ngneat/hot-toast/fesm2022/ngneat-hot-toast.mjs:6:37: ERROR: No matching export in "node_modules/@ngneat/overview/fesm2022/ngneat-overview.mjs" for import "DynamicViewModule"
    at failureErrorWithLog (D:\Angular\cloud\node_modules\vite\node_modules\esbuild\lib\main.js:1651:15)
    at D:\Angular\cloud\node_modules\vite\node_modules\esbuild\lib\main.js:1059:25
    at D:\Angular\cloud\node_modules\vite\node_modules\esbuild\lib\main.js:1527:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}

Node.js v18.17.1
  ```
  after a build command

**Comment by orestisioakeimidis** [comment]
> Can you create a stackblitz to showcase the issue?

@shhdharmen This is the stackblitz with the reproduction.

https://stackblitz-starters-ckjjdx.stackblitz.io

**Comment by Gustav0ar** [comment]
Hi @orestisioakeimidis, I was also losing my head over this issue and after a few hours debugging it, I found out the problem. Apparently the contract changed for success message. 

Just remove the function and change it a plain string.

`success: 'Fetch completed!',`

If you leave it as a function, it will pass that function when creating the dynamic component instead of the proper component and it will through that error

**Comment by orestisioakeimidis** [comment]
Hi @Gustav0ar, thanks for the debugging. I had already figured out that it works with a string, which is fine for the success notification. 

However, for the error notification, we have some additional logic to show the respective error message and we need to use a function.

@shhdharmen Any updates on that one?

**Comment by shhdharmen** [comment]
This is fixed with #155

**Comment by frct1** [comment]
> This is fixed with #155

Seems like not completely, there is warnings about unmet dependency. Anyway build leads to brake everything
```
warning " > @ngneat/hot-toast@7.0.0" has unmet peer dependency "@ngneat/overview@6.0.0".
```
UPD: Fixed with ng add

Found that '@ngneat/hot-toast' doesn't export HotToastModule (update to v7) so this is could be a reason that build doesn't work.

Node 18
ng 17
